scripts/pylib/twister/twisterlib/testinstance.py

```diff
@@ -256,21 +255,36 @@ def create_overlay(self, platform, enable_asan=False, enable_ubsan=False, enable
 
         return content
 
-    def calculate_sizes(self):
+    def calculate_sizes(self, from_buildlog: bool = False) -> SizeCalculator:
         """Get the RAM/ROM sizes of a test case.
 
         This can only be run after the instance has been executed by
         MakeGenerator, otherwise there won't be any binaries to measure.
 
         @return A SizeCalculator object
         """
+        elf_filepath = self.get_elf_file()
+        buildlog_filepath = self.get_buildlog_file() if from_buildlog else ''
+        return SizeCalculator(elf_filename=elf_filepath, extra_sections=self.testsuite.extra_sections, buildlog_filepath=buildlog_filepath)
```

Comment:
Analyzing the ELF file is still necessary because of the search for potential errors (see https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib/reports.py#L417-L421). This is the reason why we cannot exclude analyzing the ELF file. Consequently, in the scenario of using the _build.log_ file, it is necessary to pass the paths to both files.
Given the logic and architecture of Twister, the most optimal solution is to pass both paths still in the `TestInstance` class. In the `SizeCalculator` class, the existence of a path to build.log checks. If _build.log_ has been passed and exists, then the procedure of reading from _build.log_ begins.